### DISCUSSION TITLE
Is this library be used for production? | Why, what purpose, how many numbers?

### DISCUSSION THREAD

**Discussion by 1337-sys** [tulir/whatsmeow#145] [Show and tell]
Anyone using this library for production?

Does anyone in a company or otherwise use this library? Can anyone say why, for what purpose and how many numbers they use this library?

**Comment by abbasudo** [comment]
I am using this library for the company that I'm working for. we use it for CRM and support purposes. 10 numbers have registered so far and around 100 messages getting sent every day.

**Reply by muhammadrivaldy** [reply]
How many messages that you used for 1 phone number?

**Comment by moxovil** [comment]
I also use this library as an integration for CRM. Now my application works with 6500 active numbers. Sends about 1000 messages per minute. The solution is very good, the developer constantly makes updates.

**Reply by deven96** [reply]
Curious as to how you're able to scale your application

**Reply by 1337-sys** [reply]
@moxovil Which CRM?

**Reply by roniahmad** [reply]
I like to know your workaround to avoid being banned by whatsapp. Is this still valid now ?

**Reply by Mendes113** [reply]
I'm using it on a medium scale, I'd say, limiting each user to a maximum of 10 users per phone number. So far, after around two months in production, we've processed a total of approximately 200,000 messages.

**Reply by Kashishjain02** [reply]
@Mendes113 @moxovil how are you connecting multiple numbers. is there a way to run multiple sessions simultaneously

**Reply by Mendes113** [reply]
#471 
#786

**Reply by AlejoGarat** [reply]
@moxovil How do you handle the growth of whatsmeow_message_secrets? Do you periodically trim or clean it up?

**Comment by sumnerevans** [comment]
This is being used in https://github.com/mautrix/whatsapp/ which is deployed by multiple companies including Beeper and Element.

Also, this question is not appropriate for GitHub, you should ask on https://matrix.to/#/#whatsmeow:maunium.net

**Comment by 1337-sys** [comment]
@moxovil Really? 6500? How many conversations do you manage? Perfect!
